### PR TITLE
Update tungstenite

### PR DESCRIPTION
**Pull request by fabricedesre**
Update tungstenite to current version. As a nice side effect this removes a bunch of other old crates.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [X] These changes do not require tests

### CODE REVIEW

**Comment by jdm** [comment]
This looks like a lot of wss tests were failing to complete before this change, and are now completing and exposing new failures. I expect it's ok to update the expected test results.

**Comment by fabricedesre** [comment]
> This looks like a lot of wss tests were failing to complete before this change, and are now completing and exposing new failures. I expect it's ok to update the expected test results.

Is there a way to only run the ws tests locally?

**Comment by jdm** [comment]
`./mach test-wpt path/to/directory`

File: components/net/Cargo.toml [hunk 1]
```diff
@@ -61,9 +61,8 @@ servo_url = { path = "../url" }
 sha2 = "0.10"
 time = { workspace = true }
 tokio = { version = "1", package = "tokio", features = ["sync", "macros", "rt-multi-thread"] }
```
Comment (atouchet) [inline components/net/Cargo.toml:63]: Is package = "tokio" still needed here with the older version removed?

Comment (fabricedesre) [reply components/net/Cargo.toml:63]: likely not... I'm also looking at the wpt failures, so it's not ready yet.

**Comment by fabricedesre** [comment]
All tests are passing locally now.

**Review by jdm — APPROVED** [review]

**Comment by jdm** [comment]
@bors-servo r+
Thank you!

**Comment by fabricedesre** [comment]
@jdm can you merge before it bitrots again? :)

**Pull request merged**